src/main/java/org/sonatype/repository/chef/internal/proxy/ChefProxyFacetImpl.java

```diff
@@ -0,0 +1,240 @@
+    this.cookBookApiAbsoluteUrlRemover = checkNotNull(cookBookApiAbsoluteUrlRemover);
+  }
+
+  // HACK: Workaround for known CGLIB issue, forces an Import-Package for org.sonatype.nexus.repository.config
+  @Override
+  protected void doValidate(final Configuration configuration) throws Exception {
+    super.doValidate(configuration);
+  }
+
+  @Nullable
+  @Override
+  protected Content getCachedContent(final Context context) throws IOException {
+    AssetKind assetKind = context.getAttributes().require(AssetKind.class);
+    switch (assetKind) {
+      case COOKBOOK:
+        TokenMatcher.State matcherState = chefPathUtils.matcherState(context);
+        return getAsset(chefPathUtils.buildCookbookPath(matcherState));
+      case COOKBOOKS_LIST:
+      case COOKBOOK_DETAILS:
+      case COOKBOOK_DETAIL_VERSION:
+      case COOKBOOKS_SEARCH:
+        return null;
+      default:
+        throw new IllegalStateException("Received an invalid AssetKind of type: " + assetKind.name());
+    }
+  }
+
+  @Override
+  protected Content store(final Context context, final Content content) throws IOException {
+    AssetKind assetKind = context.getAttributes().require(AssetKind.class);
+    switch(assetKind) {
+      case COOKBOOK:
+        TokenMatcher.State matcherState = chefPathUtils.matcherState(context);
+        return putCookbook(content,
+            COOKBOOK,
+            chefPathUtils.buildCookbookPath(matcherState));
+      case COOKBOOK_DETAILS:
+      case COOKBOOKS_LIST:
+      case COOKBOOK_DETAIL_VERSION:
+      case COOKBOOKS_SEARCH:
+        return rewriteMetadata(content, assetKind);
+      default:
+        throw new IllegalStateException("Received an invalid AssetKind of type: " + assetKind.name());
+    }
+  }
+
+  private Content rewriteMetadata(final Content content, final AssetKind assetKind) {
+    try {
+      switch (assetKind) {
+        case COOKBOOK_DETAIL_VERSION:
+        case COOKBOOK_DETAILS:
+        case COOKBOOKS_LIST:
+        case COOKBOOKS_SEARCH:
+          return cookBookApiAbsoluteUrlRemover.maybeRewriteCookbookApiResponseAbsoluteUrls(content, assetKind);
+        default:
+          return content;
+      }
+    }
+    catch (IOException | URISyntaxException ex) {
+      log.debug("Woops " + ex.toString());
+      return content;
+    }
+  }
+
+  private Content putCookbook(final Content content,
+                              final AssetKind assetKind,
+                              final String assetPath)  throws IOException
+  {
+    StorageFacet storageFacet = facet(StorageFacet.class);
+    try (TempBlob tempBlob = storageFacet.createTempBlob(content.openInputStream(), ChefDataAccess.HASH_ALGORITHMS)) {
+      ChefAttributes chefAttributes = chefAttributeParser.getAttributesFromInputStream(tempBlob.get());
+      return doPutCookbook(chefAttributes, tempBlob, content, assetKind, assetPath);
+    }
+  }
+
+  @TransactionalStoreBlob
+  protected Content doPutCookbook(final ChefAttributes chefAttributes,
```

Comment:
Note: This is the kind of method that becomes useful for both hosted and proxy, we should start thinking about putting these into a generic facet right from the start in future.